src/main/java/com/internet/shop/dao/jdbc/ShoppingCartDaoJdbcImpl.java

```diff
@@ -0,0 +1,168 @@
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Dao
+public class ShoppingCartDaoJdbcImpl implements ShoppingCartDao {
+    @Override
+    public Optional<ShoppingCart> getByUserId(Long userId) {
+        ShoppingCart shoppingCart = null;
+        String query = "SELECT * FROM shopping_carts WHERE user_id = ? AND deleted = FALSE;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, userId);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                shoppingCart = getShoppingCartFromResultSet(resultSet);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can not get shopping cart from DB with user ID = "
+                    + userId, e);
+        }
+        shoppingCart.setProducts(getProductFromShoppingCart(shoppingCart.getId()));
+        return Optional.ofNullable(shoppingCart);
+    }
+
+    @Override
+    public ShoppingCart create(ShoppingCart item) {
+        String query = "INSERT INTO shopping_carts(user_id) VALUES (?);";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(
+                        query, Statement.RETURN_GENERATED_KEYS)) {
+            statement.setLong(1, item.getUserId());
+            statement.executeUpdate();
+            ResultSet resultSet = statement.getGeneratedKeys();
+            if (resultSet.next()) {
+                item.setId(resultSet.getLong(1));
+            }
+            return item;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Unable to create " + item, e);
+        }
+    }
+
+    @Override
+    public Optional<ShoppingCart> getById(Long item) {
+        ShoppingCart shoppingCart = null;
+        String query = "SELECT * FROM shopping_carts WHERE cart_id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, item);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                shoppingCart = getShoppingCartFromResultSet(resultSet);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("it is impossible to get a buyer's card"
+                    + " with ID = " + item, e);
+        }
+        shoppingCart.setProducts(getProductFromShoppingCart(shoppingCart.getId()));
+        return Optional.ofNullable(shoppingCart);
+    }
+
+    @Override
+    public ShoppingCart update(ShoppingCart item) {
```

Comment:
```suggestion
    public ShoppingCart update(ShoppingCart cart) {
```
same in other methods

Reply:
Done